app/services/optIn/OptInService.scala

```diff
@@ -75,6 +76,17 @@ class OptInService @Inject()(itsaStatusUpdateConnector: ITSAStatusUpdateConnecto
     }
   }
 
+  def makeOptInCall()(implicit user: MtdItUser[_],
+                      hc: HeaderCarrier,
+                      ec: ExecutionContext): Future[ITSAStatusUpdateResponse] = {
+
+    fetchSavedChosenTaxYear() flatMap {
+      case Some(intentTaxYear) => itsaStatusUpdateConnector.makeITSAStatusUpdate(taxYear = intentTaxYear, user.nino, optInUpdateReason)
```

Comment:
When we move Opt Out over to use the same code on the connector, could we make 2 wrapper methods on the connector, maybe `optOut()` and `optIn()` to keep the update reasons closer to the connector and stop them spilling out into the service?

Reply:
created sub-task for this: https://jira.tools.tax.service.gov.uk/browse/MISUV-8280